async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.validator;
+
+import static org.opensearch.sql.spark.validator.GrammarElement.ALTER_VIEW;
+import static org.opensearch.sql.spark.validator.GrammarElement.CLUSTER_BY;
+import static org.opensearch.sql.spark.validator.GrammarElement.CREATE_FUNCTION;
+import static org.opensearch.sql.spark.validator.GrammarElement.CREATE_VIEW;
+import static org.opensearch.sql.spark.validator.GrammarElement.CROSS_JOIN;
+import static org.opensearch.sql.spark.validator.GrammarElement.DESCRIBE_FUNCTION;
+import static org.opensearch.sql.spark.validator.GrammarElement.DISTRIBUTE_BY;
+import static org.opensearch.sql.spark.validator.GrammarElement.DROP_FUNCTION;
+import static org.opensearch.sql.spark.validator.GrammarElement.DROP_VIEW;
+import static org.opensearch.sql.spark.validator.GrammarElement.FILE;
+import static org.opensearch.sql.spark.validator.GrammarElement.FULL_OUTER_JOIN;
+import static org.opensearch.sql.spark.validator.GrammarElement.HINTS;
+import static org.opensearch.sql.spark.validator.GrammarElement.INLINE_TABLE;
+import static org.opensearch.sql.spark.validator.GrammarElement.INSERT;
+import static org.opensearch.sql.spark.validator.GrammarElement.LEFT_ANTI_JOIN;
+import static org.opensearch.sql.spark.validator.GrammarElement.LEFT_SEMI_JOIN;
+import static org.opensearch.sql.spark.validator.GrammarElement.LOAD;
+import static org.opensearch.sql.spark.validator.GrammarElement.MANAGE_RESOURCE;
+import static org.opensearch.sql.spark.validator.GrammarElement.MISC_FUNCTIONS;
+import static org.opensearch.sql.spark.validator.GrammarElement.REFRESH_FUNCTION;
+import static org.opensearch.sql.spark.validator.GrammarElement.REFRESH_RESOURCE;
+import static org.opensearch.sql.spark.validator.GrammarElement.RESET;
+import static org.opensearch.sql.spark.validator.GrammarElement.RIGHT_OUTER_JOIN;
+import static org.opensearch.sql.spark.validator.GrammarElement.SET;
+import static org.opensearch.sql.spark.validator.GrammarElement.SHOW_FUNCTIONS;
+import static org.opensearch.sql.spark.validator.GrammarElement.SHOW_VIEWS;
+import static org.opensearch.sql.spark.validator.GrammarElement.TABLESAMPLE;
+import static org.opensearch.sql.spark.validator.GrammarElement.TABLE_VALUED_FUNCTION;
+import static org.opensearch.sql.spark.validator.GrammarElement.TRANSFORM;
+import static org.opensearch.sql.spark.validator.GrammarElement.UDF;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+
+public class S3GlueGrammarElementValidator extends DenyListGrammarElementValidator {
```

Comment:
make it as configure file? the benfit is it could be changed without recompile code.

Reply:
That's possible, but I don't see much benefit to make it configuration file unless we make it configurable at runtime.
With class file, we can do compile time validation.

Reply:
Np: I don't see any different behavior in each subclass. Probably no need to make DenyListGrammarElementValidator abstract.

Reply:
Do you mean we don't need subclass `S3GlueGrammarElementValidator` or parent class `DenyListGrammarElementValidator`?
For S3GlueGrammarElementValidator, we can make it just an instance of DenyListGrammarElementValidator, but that would move the list of denied grammar elements into Module file, and make it harder to have unit tests for this S3Glue specific validation.